src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/OnPlatformExtension.cs

```diff
@@ -0,0 +1,73 @@
+﻿#nullable enable
+using System;
+using Avalonia.Metadata;
+using Avalonia.Platform;
+
+namespace Avalonia.Markup.Xaml.MarkupExtensions;
+
+public class OnPlatformExtension : OnPlatformExtensionBase<object, On>
+{
+    public OnPlatformExtension()
+    {
+
+    }
+
+    public OnPlatformExtension(object defaultValue)
+    {
+        Default = defaultValue;
+    }
+
+    public static bool ShouldProvideOption(IServiceProvider serviceProvider, OperatingSystemType option)
+    {
+        return serviceProvider.GetService<IRuntimePlatform>().GetRuntimeInfo().OperatingSystem == option;
+    }
+}
+
+public class OnPlatformExtension<TReturn> : OnPlatformExtensionBase<TReturn, On<TReturn>>
+{
+    public OnPlatformExtension()
+    {
+
+    }
+
+    public OnPlatformExtension(TReturn defaultValue)
+    {
+        Default = defaultValue;
+    }
+
+    public static bool ShouldProvideOption(IServiceProvider serviceProvider, OperatingSystemType option)
```

Comment:
"ShouldProvideOption" is an important method, as it will be used by developers who create custom OnPlatform-like extension, what is possible in this PR.
Do we have better ideas how to name this method?

Reply:
I'm not quite sure that it should be a _static_ method. Properties not marked by `MarkupExtensionDefaultOption` could be potentially usable by markup extension authors

Reply:
ShouldProvideOption can be both instance and static method